packages/x-charts/src/context/CartesianProvider/CartesianProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import { AxisConfig, ChartsXAxisProps, ChartsYAxisProps, ScaleName } from '../../models/axis';
+import { DatasetType } from '../../models/seriesType/config';
+import { MakeOptional } from '../../models/helpers';
+import { useDrawingArea } from '../../hooks/useDrawingArea';
+import { useSeries } from '../../hooks/useSeries';
+import { CartesianContext } from './CartesianContext';
+import { normalizeAxis } from './normalizeAxis';
+import { computeValue } from './computeValue';
+import { ExtremumGettersConfig } from '../../models';
+
+export type CartesianContextProviderProps = {
+  /**
+   * The configuration of the x-axes.
+   * If not provided, a default axis config is used.
+   * An array of [[AxisConfig]] objects.
+   */
+  xAxis?: MakeOptional<AxisConfig<ScaleName, any, ChartsXAxisProps>, 'id'>[];
+  /**
+   * The configuration of the y-axes.
+   * If not provided, a default axis config is used.
+   * An array of [[AxisConfig]] objects.
+   */
+  yAxis?: MakeOptional<AxisConfig<ScaleName, any, ChartsYAxisProps>, 'id'>[];
+  /**
+   * An array of objects that can be used to populate series and axes data using their `dataKey` property.
+   */
+  dataset?: DatasetType;
+  /**
+   * An object with x-axis extremum getters per series type.
+   */
+  xExtremumGetters: ExtremumGettersConfig;
+  /**
+   * An object with y-axis extremum getters per series type.
+   */
+  yExtremumGetters: ExtremumGettersConfig;
+  children: React.ReactNode;
+};
+
+function CartesianContextProvider(props: CartesianContextProviderProps) {
+  const prevContext = React.useContext(CartesianContext);
+
+  if (prevContext.isInitialized) {
+    return props.children;
+  }
+
+  return <CartesianContextProviderReal {...props} />;
```

Comment:
Real?

I don't understand why you need both `CartesianContextProvider` and `CartesianContextProviderReal`

Reply:
I don't see the use-case. We should not have multiple `CartesianContextProvider` otherwise we don't have a single source of truth

Reply:
I kind of disagree with this phrase. I described the changes in the description. But I'm also open to suggestions.

This was a solution I found by messing around with contexts, and it solves the issue of having to change the order of contexts, or affecting a specific context behaviour. An "issue" with our current context setup is that it is not "isolated", a lot of contexts need other contexts as well, this makes it difficult for contexts to exist on their own, and the more contexts we add, the more complex it will become.

On my solution, `Contexts` are not meant to be a "single source of truth", they are "closest source of truth" 😅, what I did was invert the rule, that instead of closest, it always respects the "farthest" or "topmost", this means we can override contexts bellow us.

The idea here is that we can redefine the pro behaviour the way we want. As long as the end signature is the same. Hence why I divided the current `CartesianContextProvider` into its important parts and copied only the required react logic into the `Pro` container.

```tsx
<CartesianContextProviderPro value={{foo:1}}>
  <CartesianContextProvider value={{foo:2}}>
    <Consumer>{useContext().foo} = 1</Consumer>
  </CartesianContextProvider>
<CartesianContextProviderPro>
```

This wouldn't be necessary if we re-export all the main components to use a "pro" version of the providers, though that would require a lot of changes in itself.